### PR TITLE
Add route tables feature & systemd-network conf file type

### PR DESCRIPTION
**Pull request by dtourde**
Add the possibility to add custom route tables.
For these new routes tables to be taken into account by systemd-networkd, I had to add `.conf` file type.

### CODE REVIEW

File: tasks/config.yml [hunk 1]
```diff
@@ -23,6 +31,34 @@
...
+    dest: "{{ systemd_networkd_conf_directory }}/{{ item.1.key }}.{{ item.0.type }}"
+    owner: root
+    group: systemd-network
+    mode: 0640
+  loop: "{{ query('subelements', all_files, 'files' ) }}"
+  loop_control:
+    label:
+      type: "{{ item.0.type }}"
+      name: "{{ item.1.key }}"
+  notify: restart systemd-networkd
```
Comment (aruhier) [inline tasks/config.yml]: I don't understand this part. Why don't you make `/etc/systemd/network` a variable, from this block line 14:

```
  ansible.builtin.template:
    src: systemd_networkd_config.j2
    dest: /etc/systemd/network/{{ item.1.key }}.{{ item.0.type }}
```

**Review by aruhier — CHANGES_REQUESTED** [review]

File: tasks/config.yml (hunk 1 above)
Comment (dtourde) [reply tasks/config.yml]: You are right, I don't remember why I did it that way, and I don't see the point having a separate task, let me change it ASAP.

Comment (dtourde) [reply tasks/config.yml]: @aruhier thanks for your patience, I implemented your recommandation.

File: tasks/config.yml [hunk 2]
```diff
@@ -7,13 +24,19 @@
...
+        dest_folder: "{{ systemd_networkd_directory }}"
       - type: netdev
         files: "{{ systemd_networkd_netdev | dict2items }}"
+        dest_folder: "{{ systemd_networkd_directory }}"
       - type: network
         files: "{{ systemd_networkd_network | dict2items }}"
+        dest_folder: "{{ systemd_networkd_directory }}"
+      - type: conf
+        files: "{{ systemd_networkd_conf | dict2items }}"
+        dest_folder: "{{ systemd_networkd_conf_directory }}"
```
Comment (aruhier) [inline tasks/config.yml:36]: Again, I'm really confused about that. Can you point me out to some docs about what would go in `/etc/systemd/networkd.conf.d`? I can't find anything about it, and I don't understand why you would need `systemd_networkd_directory` and `systemd_networkd_conf_directory`.

Comment (dtourde) [reply tasks/config.yml:36]: It's a matter of personal preference.
On this [documentation](https://www.freedesktop.org/software/systemd/man/latest/networkd.conf.html#), you can see that configuration files can be written in three places.

- `/etc/systemd/networkd.conf`
- `/etc/systemd/networkd.conf.d/*.conf`
- `/usr/lib/systemd/networkd.conf.d/*.conf`

I choose to put it in `conf.d` folder as an habit, not knowing if I would implement multiple files or not, it appears that I didn't implemented multiple conf files though.  

In the current state of the role, we can choose `/etc/systemd/networkd.conf` as the `.conf` destination with no downside.
I can implement this change if you have a preference, removing the second default variable & the `dest_folder` from the main loop.

Comment (aruhier) [reply tasks/config.yml:36]: That totally makes sense, I was confused that you used `/etc/systemd/networkd.conf.d` as an alternative to `/etc/systemd/network`. Sorry for the confusion!

Comment (dtourde) [reply tasks/config.yml:36]: No problem !
But you made a valid point, it add confusion to the role with no functionality improvement, so it make sense to simply use `/etc/systemd/network.conf`.

**Comment by aruhier** [comment]
Thanks a lot for this PR and thanks a lot for clarifying several points in the review!

**Pull request merged**